### ISSUE TITLE
copy nsec of mtime in CopyStat

### ISSUE THREAD

**Issue by JackJey** [google/brotli#891]
when using brotli command in Makefile, it doesn't build incrementally but always re-compress full files again.
this is caused by compare time difference includes nsec of mtime, current brotli cli implementation doesn't copy nsec of mtime.

https://github.com/google/brotli/blob/master/c/tools/brotli.c#L661-L662

It's welcome to include nsec of mtime when copy file stat will solve incremental build by Make. it always compress when files are updated.

**Comment by eustas** [comment]
Hmm. Let's take a look how this is done in other compressors, e.g. zlib...

**Comment by pirxpilot** [comment]
If you are here looking for help with your Makefiles this seems to do the trick:

```Makefile
%.br: %
    brotli --best --force --keep --output=$@ $<
    touch --reference $< $@
```
`touch` will update the timestamp from a source file and that will stop `make` from trying to re-brotlify the same file over and over again

And yeah - some other compressors have the same problem:
```
$ stat -c "%y %n" History.md*
2022-05-01 10:13:20.795119946 +0200 History.md
2022-05-01 10:13:20.000000000 +0200 History.md.br
2022-05-01 10:13:20.000000000 +0200 History.md.bz2
2022-05-01 10:13:20.795119946 +0200 History.md.gz
2022-05-01 10:13:20.000000000 +0200 History.md.lz
```

**Comment by eustas** [comment]
Will take a look soon. Thanks for the heads-up.

**Comment by eustas** [comment]
Should be somewhat addressed with #992
